designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+---
+parent: Administrator
+version: 4.x
+---
+
+# Bulk actions
+
+Bulk actions will be added to a number of list views in the OpenShift console. This feature will allow users to perform an action on multiple resources at once. The list of actions include adding labels, adding annotations and deleting resources.
+
+## Bulk actions in toolbar
+The bulk selector will be the first component in the toolbar on pages where bulk actions are available.
+![toolbar](img/toolbar.png)
+
+Users can click the checkbox in the bulk selector to select all items in the list. If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read ‘[#] selected’, informing users that all known items in the list are selected. Clicking the bulk selector checkbox again will deselect all items.
+![bulk-selector-states](img/bulk-selector-states.png)
+
+Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a ‘-‘ to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. ‘2 selected’). Clicking the bulk selector checkbox will deselect all items.
+![add-labels](img/add-labels.png)
+
+## Behavior
+If no items are selected, the bulk actions dropdown will be enabled, shown as a basic dropdown, but the actions within will be disabled. A tooltip will explain to users that they must select resources in order to take actions.
+![none-checked](img/none-checked.png)
+
+When items are selected, the bulk actions basic dropdown will turn into a primary dropdown, and actions within it will be enabled. 
+![add-labels](img/add-labels.png)
+
+### Item limits
+Depending on the use case, there may be scenarios where users are limited to selecting a certain number of items, assuming there is a large total in a dataset. As a general rule of thumb, the max number of items that can be selected at once will be set at 200. 
```

Comment:
Maybe an edge case, but what happens if the user has loaded over 200 items on the page and then selects the checkbox to select all? Do we notify them in any way? They may not scroll all the way down to see the tooltip on the remaining checkboxes.

Reply:
@bmignano good question - I assume clicking the select all from bulk select would only select 200 items again... so they could continue to select none/select all, only ever reaching 200 as max. Do you think we should add some sort of messaging outside hovering over non-selected checkboxes to make them aware of that?

Reply:
This will be covered in the addition of "select all on page" and "select all (max 200)" options in the bulk selector